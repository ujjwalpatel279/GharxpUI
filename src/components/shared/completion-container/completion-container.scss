// ========================================================================
// COMPONENTS/SECTIONS/COMPLETION-CONTAINER
// ========================================================================

@import '~sassHelpers';

.c-completion-container {
  background: white;
  border: 1px solid color(grey4);
  border-radius: $border-radius;
  box-shadow: $box-shadow;

  &__body {
    padding: spacer(6) spacer(4) spacer(4) spacer(4);
    text-align: center;
  }

  &__footer {
    padding: spacer(4);
    border-top: 1px solid color(grey4);
  }
}
