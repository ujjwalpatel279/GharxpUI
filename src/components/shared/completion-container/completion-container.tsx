import React, { FunctionComponent } from 'react';

import './completion-container.scss';

interface CompletionContainerProps {
  heading: string;
  text: string;
}

export const CompletionContainer: FunctionComponent<CompletionContainerProps> = ({ children, heading, text }) => {
  return (
    <section className="c-completion-container">
      <div className="c-completion-container__body">
        <h2 className="u-mb-3">{heading}</h2>
        <p>{text}</p>
      </div>

      <div className="c-completion-container__footer">{children}</div>
    </section>
  );
};
