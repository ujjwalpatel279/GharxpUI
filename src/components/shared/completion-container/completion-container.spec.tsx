import React from 'react';
import { render, cleanup } from '@testing-library/react';

import { CompletionContainer } from './completion-container';

describe('Completion Container Component', () => {
  afterEach(cleanup);

  it('renders the completion container', () => {
    const { container } = render(
      <CompletionContainer heading="Test heading" text="Test text">
        <p>I am a child</p>
      </CompletionContainer>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correct heading and text', () => {
    const { getByText } = render(
      <CompletionContainer heading="Test heading" text="Test text">
        <p>I am a child</p>
      </CompletionContainer>,
    );
    expect(getByText('Test heading')).toBeInTheDocument();
    expect(getByText('Test text')).toBeInTheDocument();
    expect(getByText('I am a child')).toBeInTheDocument();
  });
});
