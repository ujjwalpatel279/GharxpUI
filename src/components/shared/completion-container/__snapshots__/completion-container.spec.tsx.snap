// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Completion Container Component renders the completion container 1`] = `
<section
  class="c-completion-container"
>
  <div
    class="c-completion-container__body"
  >
    <h2
      class="u-mb-3"
    >
      Test heading
    </h2>
    <p>
      Test text
    </p>
  </div>
  <div
    class="c-completion-container__footer"
  >
    <p>
      I am a child
    </p>
  </div>
</section>
`;
